Wikipedia/UI-V5/WMFArticleViewController.m

```diff
@@ -491,7 +541,9 @@ - (WMFArticleReadMoreCell*)readMoreCellAtIndexPath:(NSIndexPath*)indexPath {
 
 - (CGFloat)tableView:(UITableView*)tableView heightForHeaderInSection:(NSInteger)section {
     if ([self isLeadSection:section]
-        || ([self isTOCSection:section] && section != self.indexSetOfTOCSections.firstIndex)) {
+        || ([self isTOCSection:section] && section != self.indexSetOfTOCSections.firstIndex)
+        || ![self tableView:tableView viewForHeaderInSection:section]
```

Comment:
also, this seems to only be calling `viewForHeaderInSection:` so it can indirectly run the conditional checks there. can we refactor this to use the conditional logic directly? it seems like all we really need to check is `numberOfRowsInSection == 0`, since the other two checks in `viewForHeaderInSection:` are the same as above. this is my bad, since I didn't refactor the `viewForHeaderInSection` conditionals to match these.

the way to fix all of this and DRY up the code is to refactor the 3 conditional checks into one method, and call it in both `heightForHeaderInSection` and `viewForHeaderInSection`. for example:

``` objc
 - (BOOL)shouldHideHeaderForSection:(NSUInteger)section {
  return [self isLeadSection:section] || [self isTOCSectionAfterFirst:section] || [self hasRowsInSection:section];
}

// took the liberty to factor this out too which makes the above even more expressive
- (BOOL)isTOCSectionAfterFirst:(NSUInteger)section {
  return [self isTOCSection:section] && section != self.indexSetOfTOCSections.firstIndex;
}

// same thing here, could even be a UITableView category
- (BOOL)hasRowsInSection:(NSUInteger)section {
  return [self.tableView numberOfRowsInSection:section] > 0;
}
```